crates/flox/src/commands/channel.rs

```diff
@@ -76,122 +88,131 @@ pub struct Search {
     pub async fn handle(self, flox: Flox) -> Result<()> {
         subcommand_metric!("search");
-        let channels = flox
-            .channels
-            .iter()
-            .filter_map(|entry| {
-                if HIDDEN_CHANNELS.contains_key(&*entry.from.id) {
-                    None
-                } else if DEFAULT_CHANNELS.contains_key(&*entry.from.id) {
-                    Some((ChannelType::Flox, entry))
-                } else {
-                    Some((ChannelType::User, entry))
-                }
-            })
-            .sorted_by(|a, b| Ord::cmp(a, b));
 
-        // Create `registry` parameter for `pkgdb`
-        let mut priority: Vec<String> = Vec::new();
-        let mut inputs = serde_json::Map::new();
-        for (_, entry) in channels {
-            // TODO: handle `subtrees` and `stabilities`
-            priority.push(entry.from.id.to_string());
-            inputs.insert(
-                entry.from.id.to_string(),
-                json!({"from": entry.from.to_string()}),
-            );
+        if self.search_term.is_none() {
+            bail!(
+                "'flox search' requires at least one search term, run 'flox search -h' for more information."
+            )
         }
-        let registry = json!({
-            "priority": priority,
-            "inputs": inputs,
-            "default": json!({
-                "subtrees": null, // TODO
-                "stabilities": null,  // TODO
-            })
-        });
-
-        // Create `query` parameter for `pkgdb`
-        let query = match self.search_term {
-            Some(search_term) => {
-                let mut query = serde_json::Map::new();
-                query.insert("match".to_string(), json!(search_term));
-                query
-            },
-            None => serde_json::Map::new(),
-        };
 
-        let params = json!({
-            "registry": registry,
-            "query": query,
-            // These defaults are set by `pkgdb` but are shown here
-            // for visibility.
-            // "semver": { "preferPreReleases": false }
-            // "systems": ["x86_64-linux", ...]
-        });
-
-        // TODO: I have no idea how to check verbosity here.
-        //// By default we run with `--quiet`, but if verbose is set then
-        //// stream stderr.
-        //let mut maybe_quiet: Vec<String> = Vec::new();
-        //if !self.verbose {
-        //    maybe_quiet.push("--quiet".to_string());
-        //}
-        //maybe_quiet.push("--quiet".to_string());
-        let maybe_quiet: Vec<String> = vec!["--quiet".to_string()];
+        let search_params = construct_search_params(&self.search_term, &flox)?;
+        let search_params_json = serde_json::to_string(&search_params)?;
 
         let output = Command::new(PKGDB_BIN.as_str())
             .arg("search")
-            .args(maybe_quiet)
-            .arg(params.to_string())
+            .arg("--quiet")
+            .arg(search_params_json)
```

Comment:
nit: this passing a json blob as argument interface is a bit weird 🤔

Reply:
Given the amount of information we're passing to `pkgdb` passing individual arguments would be a huge pain in the ass.

Reply:
i mean more like dumping it through stdin.
there is a max size for cli args and environment. not sure how massive these jsons get but maybe worth to keep in mind 


> The shell/OS imposed limit is usually one or two hundred thousand characters.
> 
> getconf ARG_MAX will give you the maximum input limit for a command. On the Debian system I currently have a terminal open on this returns 131072 which is 128*1024. 

https://serverfault.com/questions/163371/linux-command-line-character-limit

Reply:
We're nowhere close to 100,000 characters

Reply:
Doing a search for `hello@2.12.1` without any input channels (or constraints on them) translates to this:
```
{
    "registry": {
        "inputs": {},
        "priority": [],
        "defaults": {
            "subtrees": null,
            "stabilities": null
        }
    },
    "systems": null,
    "allow": {
        "unfree": false,
        "broken": false,
        "licenses": null
    },
    "semver": {
        "preferPreReleases": false
    },
    "query": {
        "name": null,
        "pname": null,
        "version": null,
        "semver": "2.12.1",
        "match": "hello"
    }
}
```
So a really basic search is somewhere on the order of 100 characters. I suspect a typical search will be somewhere between 100 and 1000. If this ends up being a problem we can revisit.

Reply:
all cool, to me dumping an entire document in there as an argument is just a bit weird, nothing i say we have to act on.